### PR TITLE
The rest of PR #7897 about improvement of exception message

### PR DESCRIPTION
**Pull request by stavrolia**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Non-significant (changelog entry is not needed)

### CODE REVIEW

File: dbms/src/Databases/DatabaseLazy.cpp [hunk 1]
```diff
@@ -361,8 +361,7 @@ StoragePtr DatabaseLazy::loadTable(const Context & context, const String & table
     }
     catch (const Exception & e)
     {
-        throw Exception("Cannot create table from metadata file " + table_metadata_path + ", error: " + e.displayText() +
-            ", stack trace:\n" + e.getStackTrace().toString(),
+        throw Exception("Cannot create table from metadata file " + table_metadata_path + ", error: " + e.displayText() + DB::getCurrentExceptionMessage(true),
```
Comment (alexey-milovidov) [inline dbms/src/Databases/DatabaseLazy.cpp]: If I remember correctly, `getCurrentExceptionMessage` already includes `displayText`.

Comment (alexey-milovidov) [inline dbms/src/Databases/DatabaseLazy.cpp]: The context of exception is lost.
We must:
- catch by non-const reference and call `e.addMessage` and `throw` (rethrow current exception) to save the stack trace.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
.

File: dbms/src/Databases/DatabaseLazy.cpp (hunk 1 above)
Comment (alexey-milovidov) [reply dbms/src/Databases/DatabaseLazy.cpp]: (this was a small mistake by @nikvas0 that passed review unnoticed)

Comment (stavrolia) [reply dbms/src/Databases/DatabaseLazy.cpp]: But do we want to preserve the `CANNOT_CREATE_TABLE_FROM_METADATA`?

Perhaps it's better to pass original exception to [this ctor](https://github.com/ClickHouse/ClickHouse/blob/ad2af03f989466a874757583f8bde051ae3238b3/dbms/src/Common/Exception.h#L28), such we have both correct error code and the better stack trace

File: dbms/src/Databases/DatabaseOrdinary.cpp [hunk 2]
```diff
@@ -81,8 +81,7 @@ try
 catch (const Exception & e)
 {
     throw Exception(
-        "Cannot create object '" + query.table + "' from query " + serializeAST(query) + ", error: " + e.displayText() + ", stack trace:\n"
-            + e.getStackTrace().toString(),
+        "Cannot create object '" + query.table + "' from query " + serializeAST(query) + ". Error: " + DB::getCurrentExceptionMessage(true),
```
Comment (alexey-milovidov) [inline dbms/src/Databases/DatabaseOrdinary.cpp]: Exception context is lost.

File: dbms/src/Databases/DatabaseOrdinary.cpp [hunk 3]
```diff
@@ -138,8 +137,7 @@ void DatabaseOrdinary::loadStoredObjects(
         catch (const Exception & e)
         {
             throw Exception(
-                "Cannot parse definition from metadata file " + full_path + ", error: " + e.displayText() + ", stack trace:\n"
-                    + e.getStackTrace().toString(), ErrorCodes::CANNOT_PARSE_TEXT);
+                "Cannot parse definition from metadata file " + full_path + ". Error: " + DB::getCurrentExceptionMessage(true), ErrorCodes::CANNOT_PARSE_TEXT);
```
Comment (alexey-milovidov) [inline dbms/src/Databases/DatabaseOrdinary.cpp]: Exception context is lost.

File: dbms/src/Databases/DatabaseLazy.cpp (hunk 1 above)
Comment (alexey-milovidov) [reply dbms/src/Databases/DatabaseLazy.cpp]: Yes, it can be better.

Comment (alexey-milovidov) [reply dbms/src/Databases/DatabaseLazy.cpp]: But it's not necessary.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**